docs/user-guide/src/bmo/firmware_updates.md

```diff
@@ -0,0 +1,155 @@
+  uid: 1f9d5b76-5b17-44a1-84f8-7242daafc51d
+spec:
+  updates: []
+status:
+  components:
+  - component: bios
+    currentVersion: 2.3.5
+    initialVersion: 2.3.5
+  - component: bmc
+    currentVersion: 6.10.30.00
+    initialVersion: 6.10.30.00
+  conditions:
+  - lastTransitionTime: "2024-08-08T16:44:35Z"
+    message: ""
+    observedGeneration: 1
+    reason: OK
+    status: "True"
+    type: Valid
+  - lastTransitionTime: "2024-08-08T16:44:35Z"
+    message: ""
+    observedGeneration: 1
+    reason: OK
+    status: "False"
+    type: ChangeDetected
+  lastUpdated: "2024-08-08T16:44:35Z"
+```
+
+This example was taken from a real hardware and was automatically generated:
+
+- The `spec.updates` list is empty - no change is requested by the user.
+
+- The `status.updates` will only be present when `spec.updates` is not empty
+  and an update was executed.
+
+- The `status.components` information is populated with the current values
+  detected by Ironic. If an update is executed, the updated information will
+  be available when the host transitions from `available` state.
+
+- The `Valid` condition is `True`, which means that `spec.updates` are valid,
+  since it was automatically generated. We allow the `spec.updates` to be an
+  empty list. The condition will be set to `False` if any value in
+  `spec.updates` fails validation.
+
+- The `ChangeDetected` condition is `False`, which means that the information
+  provided in status matches the information from Ironic and from
+  `spec.updates`. This condition will be set to `True` after you modify
+  `spec.updates` until the change is reflected in `status.updates`.
+
+**Warning:** The components in status are only updated on
+enrollment and provisioning. We do not periodically retrieve
+firmware versions unless an update is executed.
+
+**Note:** When manually creating the `HostFirmwareComponent` resource,
+the information for `status` and `metadata` will be updated during
+`inspecting`.
+
+## How to change firmware components
+
+To change one or more components for a host, update the corresponding
+`HostFirmwareComponents` resource, changing or adding the required components
+to `spec.updates`. For example:
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: HostFirmwareComponents
+metadata:
+  name: worker-0
+  namespace: my-cluster
+  # ...
+spec:
+  updates:
+  - component: <bmc or bios>
+    url: https://newfirmwareforcomponent/file
+status:
+  # ...
+```
+
+The firmware update for the components are only executed when the host is in
+`preparing` state. When adding a new `BareMetalHost` and manually creating the
+`HostFirmwareComponents` resource for it, you can specify the updates that
+must occur for that host before it goes to `available`.
+
+In case you have a host that is `provisioned`, and you would like to execute a
+firmware update, you will need to edit the `HostFirmwareComponents` CR and
+the host will be deprovisioned so it can go to `preparing` to execute the
```

Comment:
This sentence is confusing: I don't think we will automatically deprovision the host? I think you wanted to say something along the lines of "you will need to edit .... and then trigger deprovisioning so it can go to `preparing`" (make the word deprovisioning a link to provisioning.md